.changeset/spicy-toes-confess.md

```diff
@@ -0,0 +1,5 @@
+---
+"@changesets/action": minor
+---
+
+Allow to specify `createGithubReleases: aggregate`, in order to publish a single GitHub Release
```

Comment:
My first idea when thinking about this feature was for this to be controlled by the `.changeset/config.json` and to **also** generate a single `CHANGELOG.md` for the involved packages. I have also wanted to only support this (at least at the start) for linked/fixed packages - as everything that gets released when using them shares a version so it simplifies `tag` and releases name creation.

What do you think about that idea? Any pros/cons that come to your mind when considering both approaches?

Reply:
That's an interesting take. I was thinking about this with our team, and our biggest issue is the Releases on GitHub that's being bloated. Using only a single Release makes perfect sense for us (similar to the single PR created by Changesets). 

Regarding the `CHANGELOG`, that's actually a different feature, no? I mean, you can still use aggregated release either with a unified `CHANGELOG`, or separate (per-package). 
My take on `CHANGELOG` is that I like it within each package and not in a single place, because this way it's being part of the NPM packages and other tools (like Npm, or NPM stats services) can use it, and it's easier to have the scope of changes per each package in a single place. Usually if you are having an issue with a library, you'll look for the changes in that library/package specifically, and it's easier to find when it's located near the code. 

I can see where merging `CHANGELOG`s might be useful for sure, especially if you are releasing all packages within a repo together (we did that in some libraries in the past, I think Babel is doing that?)
I guess this feature belongs to the `cli` more than the GH Action?